packages/i18n-calypso/types/index.d.ts

```diff
@@ -98,15 +103,27 @@ declare namespace i18nCalypso {
 
 	export function useTranslate(): typeof translate;
 
+	export function reRenderTranslations(): void;
+
 	export type TranslateHook = (
 		translation: TranslateResult,
 		options: NormalizedTranslateArgs
 	) => TranslateResult;
-
 	export function registerTranslateHook( hook: TranslateHook ): void;
 
+	export type ComponentUpdateHook = ( ...args: any ) => any;
+	export function registerComponentUpdateHook( hook: ComponentUpdateHook ): void;
+
+	export function getLocale(): LocaleData;
 	export function getLocaleSlug(): string | null;
+	export function getLocaleVariant(): string | undefined;
+	export function isRtl(): boolean;
 	export const defaultLocaleSlug: string;
+
+	export type EventListener = ( ...payload: any ) => any;
+	export function on( eventName: string, listener: EventListener ): void;
+	export function off( eventName: string, listener: EventListener ): void;
+	export function emit( eventName: string, ...payload: any ): void;
```

Comment:
Are there specific event names that could be added here or are they generated somehow?

Reply:
There's only one ( `change`  ) event that's currently being dispatched from within the package. However, since `emit` is also a public method, that means that consumer packages could practically dispatch any event name.

Reply:
Ah, of course, that makes sense.